homotopy-graphics/src/svg.rs

```diff
@@ -1,2 +1,51 @@
+use std::fmt::Write;
+
+use homotopy_core::Generator;
+
+use crate::style::{Color, GeneratorStyle, SignatureStyleData};
+
 pub mod render;
 pub mod shape;
+
+pub fn stylesheet(styles: &impl SignatureStyleData, prefix: &str) -> String {
+    let mut stylesheet = String::new();
+
+    for generator in styles.generators() {
+        let style = styles.generator_style(generator).unwrap();
+
+        writeln!(
+            stylesheet,
+            ".{name} {{ fill: {color}; stroke: {color}; }}",
+            name = class(prefix, generator, "surface"),
+            color = hex(&style.color().lighten(0.1))
+        )
+        .unwrap();
+        writeln!(
+            stylesheet,
+            ".{name} {{ stroke: {color}; }}",
+            name = class(prefix, generator, "wire"),
+            color = hex(&style.color().lighten(0.05))
+        )
+        .unwrap();
+        writeln!(
+            stylesheet,
+            ".{name} {{ fill: {color}; }}",
+            name = class(prefix, generator, "point"),
+            color = hex(&style.color())
+        )
+        .unwrap();
+    }
+
+    stylesheet
+}
+
+pub fn class(prefix: &str, generator: Generator, suffix: &str) -> String {
+    format!(
+        "{}__{}-{}--{}",
+        prefix, generator.id, generator.dimension, suffix
+    )
+}
+
+fn hex(color: &Color) -> String {
```

Comment:
Could we move this into `Color`? Since it seems we use it in quite a few places.

Reply:
For now, I have switched to using `to_string` calls instead. This results in slightly less code but if we ever change the way we `fmt::Display` colors, it could break some of the styles.

Reply:
That's fine. Maybe write a comment so we don't forget.

Reply:
Great.